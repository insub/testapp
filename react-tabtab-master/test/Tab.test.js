import React from 'react';
import Tab, {TabStyle} from '../src/Tab';
import {shallow, mount} from 'enzyme';
import 'jest-styled-components'

describe('Tab', () => {
  describe('render TabList', () => {
    it('render pure text', () => {
      const component = mount(
        <Tab>
          tab
        </Tab>
      );
      expect(component.html()).toMatchSnapshot();
    })

    it('render component.', () => {
      const component = mount(
        <Tab>
          <span>
            <i className="fa fa-icon"></i>
            tab
          </span>
        </Tab>
      );
      expect(component.html()).toMatchSnapshot();
    })
  })

  describe('event', () => {
    it('onClick', () => {
      const mockHandleClick = jest.fn();
      const tabKey = 1;
      const component = mount(
        <Tab index={tabKey}
             handleTabChange={mockHandleClick}>
          text
        </Tab>
      );
      component.find('Tab').simulate('click');
      expect(mockHandleClick).toBeCalled();
      expect(mockHandleClick.mock.calls[0][0]).toEqual(tabKey);
    })
  })

  test('custom ListStyle style', () => {
    const CustomTabStyle = TabStyle.extend`
      background-color: red;
    `;
    const component = shallow(
      <Tab CustomTabStyle={CustomTabStyle}>
        <span>tab</span>
      </Tab>
    );
    expect(component).toHaveStyleRule('background-color', 'red');  
  })
})

