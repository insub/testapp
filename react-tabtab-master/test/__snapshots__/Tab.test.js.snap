// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab render TabList render component. 1`] = `"<li class=\\"sc-bwzfXH jsVSXf\\" role=\\"tab\\" id=\\"react-tabtab-tab-undefined\\" aria-controls=\\"react-tabtab-panel-undefined\\"><span><i class=\\"fa fa-icon\\"></i>tab</span></li>"`;

exports[`Tab render TabList render pure text 1`] = `"<li class=\\"sc-bwzfXH jsVSXf\\" role=\\"tab\\" id=\\"react-tabtab-tab-undefined\\" aria-controls=\\"react-tabtab-panel-undefined\\">tab</li>"`;
